File: app/src/main/java/com/example/ui/screens/SettingsScreen.kt
package com.example.ui.screens

import android.annotation.SuppressLint
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.ExperimentalLayoutApi
import androidx.compose.foundation.layout.FlowRow
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.automirrored.filled.ArrowForwardIos
import androidx.compose.material.icons.filled.AutoStories
import androidx.compose.material.icons.filled.Close
import androidx.compose.material.icons.filled.Notes
import androidx.compose.material.icons.filled.RestartAlt
import androidx.compose.material3.AssistChip
import androidx.compose.material3.AssistChipDefaults
import androidx.compose.material3.Button
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.OutlinedTextFieldDefaults
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Surface
import androidx.compose.material3.Switch
import androidx.compose.material3.SwitchDefaults
import androidx.compose.material3.Text
import androidx.compose.material3.TopAppBar
import androidx.compose.material3.TopAppBarDefaults
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.data.ClockSettings
import com.example.ui.components.ClockDisplay
import java.time.LocalDateTime

/**
 * SettingsScreen
 *
 * Configures all functional clock behaviors:
 * 1. Time Format (12-hour vs 24-hour)
 * 2. Show Seconds (ON/OFF)
 * 3. Show Date (ON/OFF)
 * 4. Custom Note (Message along with clock, toggle, presets)
 * 5. Clock Size (Small, Medium, Large, Extra Large)
 * 6. Clock Position (Top, Center, Bottom)
 * 7. Reset to Defaults
 */
@OptIn(ExperimentalMaterial3Api::class, ExperimentalLayoutApi::class)
@SuppressLint("NewApi")
@Composable
fun SettingsScreen(
    dateTime: LocalDateTime,
    settings: ClockSettings,
    on24HourToggle: (Boolean) -> Unit,
    onSecondsToggle: (Boolean) -> Unit,
    onDateToggle: (Boolean) -> Unit,
    onCustomNoteChange: (String) -> Unit,
    onShowNoteToggle: (Boolean) -> Unit,
    onSizeSelected: (String) -> Unit,
    onPositionSelected: (String) -> Unit,
    onResetDefaults: () -> Unit,
    onNavigateHistory: () -> Unit = {},
    onBack: () -> Unit,
    modifier: Modifier = Modifier
) {
    Scaffold(
        modifier = modifier.fillMaxSize(),
        containerColor = Color(0xFF0D1117),
        topBar = {
            TopAppBar(
                title = {
                    Text(
                        text = "Time & Clock Settings",
                        color = Color.White,
                        fontWeight = FontWeight.Bold
                    )
                },
                navigationIcon = {
                    IconButton(
                        onClick = onBack,
                        modifier = Modifier.testTag("settings_back_btn")
                    ) {
                        Icon(
                            imageVector = Icons.AutoMirrored.Filled.ArrowBack,
                            contentDescription = "Back",
                            tint = Color.White
                        )
                    }
                },
                colors = TopAppBarDefaults.topAppBarColors(
                    containerColor = Color(0xFF0D1117)
                )
            )
        }
    ) { innerPadding ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(innerPadding)
                .padding(horizontal = 16.dp)
                .verticalScroll(rememberScrollState()),
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            // Live Preview Card (Immediate visual feedback upon toggle)
            Card(
                modifier = Modifier
                    .fillMaxWidth()
                    .height(170.dp),
                shape = RoundedCornerShape(20.dp),
                colors = CardDefaults.cardColors(
                    containerColor = Color.Black.copy(alpha = 0.4f)
                )
            ) {
                Box(
                    modifier = Modifier
                        .fillMaxSize()
                        .padding(12.dp),
                    contentAlignment = Alignment.Center
                ) {
                    ClockDisplay(
                        dateTime = dateTime,
                        settings = settings,
                        isFullScreen = false,
                        previewScale = 0.75f
                    )
                }
            }

            // Section 1: Time Format (12h vs 24h)
            Card(
                modifier = Modifier.fillMaxWidth(),
                shape = RoundedCornerShape(18.dp),
                colors = CardDefaults.cardColors(containerColor = Color.White.copy(alpha = 0.05f))
            ) {
                Column(
                    modifier = Modifier.padding(16.dp),
                    verticalArrangement = Arrangement.spacedBy(12.dp)
                ) {
                    Text(
                        text = "Time Format",
                        style = MaterialTheme.typography.titleMedium,
                        fontWeight = FontWeight.Bold,
                        color = Color.White
                    )

                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.spacedBy(10.dp)
                    ) {
                        SegmentOption(
                            label = "12-Hour (AM/PM)",
                            isSelected = !settings.is24Hour,
                            accentColor = Color(settings.colorHex),
                            onClick = { on24HourToggle(false) },
                            modifier = Modifier.weight(1f)
                        )
                        SegmentOption(
                            label = "24-Hour",
                            isSelected = settings.is24Hour,
                            accentColor = Color(settings.colorHex),
                            onClick = { on24HourToggle(true) },
                            modifier = Modifier.weight(1f)
                        )
                    }
                }
            }

            // Section 2: Display Toggles (Seconds & Date)
            Card(
                modifier = Modifier.fillMaxWidth(),
                shape = RoundedCornerShape(18.dp),
                colors = CardDefaults.cardColors(containerColor = Color.White.copy(alpha = 0.05f))
            ) {
                Column(modifier = Modifier.padding(16.dp)) {
                    // Seconds Switch
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.SpaceBetween,
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Column {
                            Text(
                                text = "Show Seconds",
                                style = MaterialTheme.typography.titleSmall,
                                fontWeight = FontWeight.SemiBold,
                                color = Color.White
                            )
                            Text(
                                text = "Displays :SS on the clock face",
                                style = MaterialTheme.typography.bodySmall,
                                color = Color.White.copy(alpha = 0.6f)
                            )
                        }

                        Switch(
                            checked = settings.showSeconds,
                            onCheckedChange = onSecondsToggle,
                            colors = SwitchDefaults.colors(
                                checkedThumbColor = Color.White,
                                checkedTrackColor = Color(settings.colorHex)
                            ),
                            modifier = Modifier.testTag("switch_show_seconds")
                        )
                    }

                    Spacer(modifier = Modifier.height(14.dp))

                    // Date Switch
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.SpaceBetween,
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Column {
                            Text(
                                text = "Show Date",
                                style = MaterialTheme.typography.titleSmall,
                                fontWeight = FontWeight.SemiBold,
                                color = Color.White
                            )
                            Text(
                                text = "Displays Day, Month, Date, Year",
                                style = MaterialTheme.typography.bodySmall,
                                color = Color.White.copy(alpha = 0.6f)
                            )
                        }

                        Switch(
                            checked = settings.showDate,
                            onCheckedChange = onDateToggle,
                            colors = SwitchDefaults.colors(
                                checkedThumbColor = Color.White,
                                checkedTrackColor = Color(settings.colorHex)
                            ),
                            modifier = Modifier.testTag("switch_show_date")
                        )
                    }
                }
            }

            // Section 3: Custom Note
            Card(
                modifier = Modifier
                    .fillMaxWidth()
                    .testTag("card_custom_note"),
                shape = RoundedCornerShape(18.dp),
                colors = CardDefaults.cardColors(containerColor = Color.White.copy(alpha = 0.05f))
            ) {
                Column(
                    modifier = Modifier.padding(16.dp),
                    verticalArrangement = Arrangement.spacedBy(14.dp)
                ) {
                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.SpaceBetween,
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Column {
                            Text(
                                text = "Custom Note",
                                style = MaterialTheme.typography.titleMedium,
                                fontWeight = FontWeight.Bold,
                                color = Color.White
                            )
                            Text(
                                text = "Display a personal note or reminder with the clock",
                                style = MaterialTheme.typography.bodySmall,
                                color = Color.White.copy(alpha = 0.6f)
                            )
                        }

                        Switch(
                            checked = settings.showNote,
                            onCheckedChange = onShowNoteToggle,
                            colors = SwitchDefaults.colors(
                                checkedThumbColor = Color.White,
                                checkedTrackColor = Color(settings.colorHex)
                            ),
                            modifier = Modifier.testTag("switch_show_note")
                        )
                    }

                    // Text Input for Note
                    OutlinedTextField(
                        value = settings.customNote,
                        onValueChange = onCustomNoteChange,
                        modifier = Modifier
                            .fillMaxWidth()
                            .testTag("input_custom_note"),
                        placeholder = {
                            Text(
                                text = "Enter a custom note (e.g. Focus & Win)",
                                color = Color.White.copy(alpha = 0.4f)
                            )
                        },
                        leadingIcon = {
                            Icon(
                                imageVector = Icons.Default.Notes,
                                contentDescription = "Note Icon",
                                tint = Color(settings.colorHex)
                            )
                        },
                        trailingIcon = {
                            if (settings.customNote.isNotEmpty()) {
                                IconButton(
                                    onClick = { onCustomNoteChange("") },
                                    modifier = Modifier.testTag("clear_note_btn")
                                ) {
                                    Icon(
                                        imageVector = Icons.Default.Close,
                                        contentDescription = "Clear",
                                        tint = Color.White.copy(alpha = 0.6f)
                                    )
                                }
                            }
                        },
                        singleLine = true,
                        shape = RoundedCornerShape(12.dp),
                        colors = OutlinedTextFieldDefaults.colors(
                            focusedTextColor = Color.White,
                            unfocusedTextColor = Color.White,
                            focusedBorderColor = Color(settings.colorHex),
                            unfocusedBorderColor = Color.White.copy(alpha = 0.2f),
                            cursorColor = Color(settings.colorHex),
                            focusedContainerColor = Color.Black.copy(alpha = 0.25f),
                            unfocusedContainerColor = Color.Black.copy(alpha = 0.15f)
                        )
                    )

                    // Quick Note Presets
                    Text(
                        text = "Quick Presets",
                        style = MaterialTheme.typography.labelMedium,
                        color = Color.White.copy(alpha = 0.7f),
                        fontWeight = FontWeight.SemiBold
                    )

                    FlowRow(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.spacedBy(8.dp),
                        verticalArrangement = Arrangement.spacedBy(8.dp)
                    ) {
                        val presets = listOf(
                            "Focus Time 🎯",
                            "Stay Hydrated 💧",
                            "Study Session 📚",
                            "Meeting Soon 💼",
                            "Breathe & Relax 🌿",
                            "Carpe Diem ✨"
                        )
                        presets.forEach { preset ->
                            val isSelected = settings.customNote == preset
                            AssistChip(
                                onClick = { onCustomNoteChange(preset) },
                                label = {
                                    Text(
                                        text = preset,
                                        fontSize = 12.sp,
                                        fontWeight = if (isSelected) FontWeight.Bold else FontWeight.Normal,
                                        color = if (isSelected) Color.Black else Color.White.copy(alpha = 0.85f)
                                    )
                                },
                                colors = AssistChipDefaults.assistChipColors(
                                    containerColor = if (isSelected) Color(settings.colorHex) else Color.White.copy(alpha = 0.08f)
                                ),
                                border = if (isSelected) null else androidx.compose.foundation.BorderStroke(1.dp, Color.White.copy(alpha = 0.15f)),
                                shape = RoundedCornerShape(10.dp),
                                modifier = Modifier.testTag("preset_chip_$preset")
                            )
                        }
                    }
                }
            }

            // Section 4: Clock Size
            Card(
                modifier = Modifier.fillMaxWidth(),
                shape = RoundedCornerShape(18.dp),
                colors = CardDefaults.cardColors(containerColor = Color.White.copy(alpha = 0.05f))
            ) {
                Column(
                    modifier = Modifier.padding(16.dp),
                    verticalArrangement = Arrangement.spacedBy(12.dp)
                ) {
                    Text(
                        text = "Clock Size",
                        style = MaterialTheme.typography.titleMedium,
                        fontWeight = FontWeight.Bold,
                        color = Color.White
                    )

                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.spacedBy(8.dp)
                    ) {
                        listOf(
                            ClockSettings.SIZE_SMALL to "Small",
                            ClockSettings.SIZE_MEDIUM to "Medium",
                            ClockSettings.SIZE_LARGE to "Large",
                            ClockSettings.SIZE_XLARGE to "XL"
                        ).forEach { (sizeKey, label) ->
                            SegmentOption(
                                label = label,
                                isSelected = settings.clockSize == sizeKey,
                                accentColor = Color(settings.colorHex),
                                onClick = { onSizeSelected(sizeKey) },
                                modifier = Modifier.weight(1f)
                            )
                        }
                    }
                }
            }

            // Section 4: Clock Position
            Card(
                modifier = Modifier.fillMaxWidth(),
                shape = RoundedCornerShape(18.dp),
                colors = CardDefaults.cardColors(containerColor = Color.White.copy(alpha = 0.05f))
            ) {
                Column(
                    modifier = Modifier.padding(16.dp),
                    verticalArrangement = Arrangement.spacedBy(12.dp)
                ) {
                    Text(
                        text = "Clock Position",
                        style = MaterialTheme.typography.titleMedium,
                        fontWeight = FontWeight.Bold,
                        color = Color.White
                    )

                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        horizontalArrangement = Arrangement.spacedBy(8.dp)
                    ) {
                        listOf(
                            ClockSettings.POS_TOP to "Top",
                            ClockSettings.POS_CENTER to "Center",
                            ClockSettings.POS_BOTTOM to "Bottom"
                        ).forEach { (posKey, label) ->
                            SegmentOption(
                                label = label,
                                isSelected = settings.clockPosition == posKey,
                                accentColor = Color(settings.colorHex),
                                onClick = { onPositionSelected(posKey) },
                                modifier = Modifier.weight(1f)
                            )
                        }
                    }
                }
            }

            // Section 5: Reading Sessions & History
            Card(
                onClick = onNavigateHistory,
                modifier = Modifier
                    .fillMaxWidth()
                    .testTag("settings_reading_history_card"),
                shape = RoundedCornerShape(16.dp),
                colors = CardDefaults.cardColors(
                    containerColor = Color.White.copy(alpha = 0.05f)
                )
            ) {
                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(16.dp),
                    verticalAlignment = Alignment.CenterVertically,
                    horizontalArrangement = Arrangement.SpaceBetween
                ) {
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Surface(
                            shape = RoundedCornerShape(10.dp),
                            color = Color(settings.colorHex).copy(alpha = 0.15f),
                            modifier = Modifier.size(38.dp)
                        ) {
                            Box(contentAlignment = Alignment.Center) {
                                Icon(
                                    imageVector = Icons.Default.AutoStories,
                                    contentDescription = null,
                                    tint = Color(settings.colorHex),
                                    modifier = Modifier.size(20.dp)
                                )
                            }
                        }
                        Spacer(modifier = Modifier.width(12.dp))
                        Column {
                            Text(
                                text = "Reading & Focus History",
                                fontWeight = FontWeight.Bold,
                                fontSize = 14.sp,
                                color = Color.White
                            )
                            Text(
                                text = "View recorded sessions & manage history",
                                fontSize = 12.sp,
                                color = Color.White.copy(alpha = 0.55f)
                            )
                        }
                    }
                    Icon(
                        imageVector = Icons.AutoMirrored.Filled.ArrowForwardIos,
                        contentDescription = "Open History",
                        tint = Color.White.copy(alpha = 0.4f),
                        modifier = Modifier.size(14.dp)
                    )
                }
            }

            // Section 6: Reset Defaults
            OutlinedButton(
                onClick = onResetDefaults,
                modifier = Modifier
                    .fillMaxWidth()
                    .height(50.dp)
                    .testTag("reset_defaults_btn"),
                shape = RoundedCornerShape(14.dp),
                colors = ButtonDefaults.outlinedButtonColors(
                    contentColor = Color(0xFFFF453A)
                ),
                border = androidx.compose.foundation.BorderStroke(1.dp, Color(0xFFFF453A).copy(alpha = 0.6f))
            ) {
                Icon(
                    imageVector = Icons.Default.RestartAlt,
                    contentDescription = "Reset",
                    tint = Color(0xFFFF453A)
                )
                Spacer(modifier = Modifier.width(8.dp))
                Text(
                    text = "Reset All Settings to Defaults",
                    fontWeight = FontWeight.SemiBold
                )
            }

            Spacer(modifier = Modifier.height(20.dp))
        }
    }
}

@Composable
private fun SegmentOption(
    label: String,
    isSelected: Boolean,
    accentColor: Color,
    onClick: () -> Unit,
    modifier: Modifier = Modifier
) {
    Surface(
        onClick = onClick,
        modifier = modifier
            .height(44.dp)
            .testTag("seg_opt_$label"),
        shape = RoundedCornerShape(12.dp),
        color = if (isSelected) accentColor else Color.White.copy(alpha = 0.08f),
        border = if (isSelected) null else androidx.compose.foundation.BorderStroke(1.dp, Color.White.copy(alpha = 0.1f))
    ) {
        Box(contentAlignment = Alignment.Center) {
            Text(
                text = label,
                style = MaterialTheme.typography.labelMedium,
                fontWeight = if (isSelected) FontWeight.Bold else FontWeight.Medium,
                color = if (isSelected) {
                    if (isLightColor(accentColor)) Color.Black else Color.White
                } else {
                    Color.White.copy(alpha = 0.85f)
                }
            )
        }
    }
}

File: flutter_project/lib/screens/settings_screen.dart
import 'package:flutter/material.dart';
import '../models/clock_settings.dart';
import '../widgets/clock_display.dart';
import '../widgets/setting_tile.dart';

/// SettingsScreen
///
/// Educational Note:
/// - Controls functional time parameters and geometry layout.
/// - Demonstrates state propagation: calling onUpdateSettings(newSettings)
///   immediately updates the parent state, which triggers a rebuild across all widgets.
class SettingsScreen extends StatelessWidget {
  final DateTime currentTime;
  final ClockSettings settings;
  final Function(ClockSettings) onUpdateSettings;

  const SettingsScreen({
    super.key,
    required this.currentTime,
    required this.settings,
    required this.onUpdateSettings,
  });

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: const Color(0xFF0D1117),
      appBar: AppBar(
        title: const Text('Time & Clock Settings', style: TextStyle(fontWeight: FontWeight.bold)),
        backgroundColor: const Color(0xFF0D1117),
        elevation: 0,
      ),
      body: ListView(
        padding: const EdgeInsets.all(16.0),
        children: [
          // Live Clock Preview
          Container(
            height: 150,
            padding: const EdgeInsets.all(16),
            decoration: BoxDecoration(
              color: Colors.black.withOpacity(0.4),
              borderRadius: BorderRadius.circular(20),
            ),
            child: Center(
              child: ClockDisplay(
                dateTime: currentTime,
                settings: settings,
                isFullScreen: false,
                previewScale: 0.72,
              ),
            ),
          ),
          const SizedBox(height: 16),

          // 1. Time Format Setting
          SettingCard(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                const Text(
                  'Time Format',
                  style: TextStyle(color: Colors.white, fontSize: 16, fontWeight: FontWeight.bold),
                ),
                const SizedBox(height: 12),
                Row(
                  children: [
                    Expanded(
                      child: _ChoiceChip(
                        label: '12-Hour (AM/PM)',
                        isSelected: !settings.is24Hour,
                        accentColor: settings.clockColor,
                        onTap: () {
                          onUpdateSettings(settings.copyWith(is24Hour: false));
                        },
                      ),
                    ),
                    const SizedBox(width: 8),
                    Expanded(
                      child: _ChoiceChip(
                        label: '24-Hour',
                        isSelected: settings.is24Hour,
                        accentColor: settings.clockColor,
                        onTap: () {
                          onUpdateSettings(settings.copyWith(is24Hour: true));
                        },
                      ),
                    ),
                  ],
                ),
              ],
            ),
          ),
          const SizedBox(height: 12),

          // 2. Seconds and Date Toggles
          SettingCard(
            child: Column(
              children: [
                SwitchListTile(
                  contentPadding: EdgeInsets.zero,
                  title: const Text('Show Seconds', style: TextStyle(color: Colors.white, fontWeight: FontWeight.bold)),
                  subtitle: const Text('Display :SS on clock', style: TextStyle(color: Colors.white60, fontSize: 12)),
                  value: settings.showSeconds,
                  activeColor: settings.clockColor,
                  onChanged: (val) {
                    onUpdateSettings(settings.copyWith(showSeconds: val));
                  },
                ),
                const Divider(color: Colors.white10),
                SwitchListTile(
                  contentPadding: EdgeInsets.zero,
                  title: const Text('Show Date', style: TextStyle(color: Colors.white, fontWeight: FontWeight.bold)),
                  subtitle: const Text('Display Day, Month, Date, Year', style: TextStyle(color: Colors.white60, fontSize: 12)),
                  value: settings.showDate,
                  activeColor: settings.clockColor,
                  onChanged: (val) {
                    onUpdateSettings(settings.copyWith(showDate: val));
                  },
                ),
              ],
            ),
          ),
          const SizedBox(height: 12),

          // 3. Custom Note Setting
          SettingCard(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Row(
                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
                  children: [
                    const Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        Text(
                          'Custom Note',
                          style: TextStyle(color: Colors.white, fontSize: 16, fontWeight: FontWeight.bold),
                        ),
                        SizedBox(height: 2),
                        Text(
                          'Show a personal note with the clock',
                          style: TextStyle(color: Colors.white54, fontSize: 13),
                        ),
                      ],
                    ),
                    Switch(
                      value: settings.showNote,
                      activeColor: settings.clockColor,
                      onChanged: (val) {
                        onUpdateSettings(settings.copyWith(showNote: val));
                      },
                    ),
                  ],
                ),
                const SizedBox(height: 12),
                TextFormField(
                  initialValue: settings.customNote,
                  style: const TextStyle(color: Colors.white),
                  decoration: InputDecoration(
                    hintText: 'Enter custom note...',
                    hintStyle: const TextStyle(color: Colors.white38),
                    prefixIcon: Icon(Icons.note_alt_outlined, color: settings.clockColor),
                    filled: true,
                    fillColor: Colors.black26,
                    border: OutlineInputBorder(
                      borderRadius: BorderRadius.circular(12),
                      borderSide: BorderSide(color: settings.clockColor.withOpacity(0.3)),
                    ),
                    focusedBorder: OutlineInputBorder(
                      borderRadius: BorderRadius.circular(12),
                      borderSide: BorderSide(color: settings.clockColor),
                    ),
                  ),
                  onChanged: (val) {
                    onUpdateSettings(settings.copyWith(customNote: val));
                  },
                ),
                const SizedBox(height: 10),
                Wrap(
                  spacing: 6,
                  runSpacing: 6,
                  children: [
                    'Focus Time 🎯',
                    'Stay Hydrated 💧',
                    'Study Session 📚',
                    'Be Present ✨',
                  ].map((preset) {
                    final isSelected = settings.customNote == preset;
                    return ActionChip(
                      label: Text(
                        preset,
                        style: TextStyle(
                          fontSize: 12,
                          color: isSelected ? Colors.black : Colors.white,
                          fontWeight: isSelected ? FontWeight.bold : FontWeight.normal,
                        ),
                      ),
                      backgroundColor: isSelected ? settings.clockColor : Colors.white.withOpacity(0.1),
                      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(10)),
                      onPressed: () {
                        onUpdateSettings(settings.copyWith(customNote: preset));
                      },
                    );
                  }).toList(),
                ),
              ],
            ),
          ),
          const SizedBox(height: 12),

          // 4. Clock Size
          SettingCard(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                const Text(
                  'Clock Size',
                  style: TextStyle(color: Colors.white, fontSize: 16, fontWeight: FontWeight.bold),
                ),
                const SizedBox(height: 12),
                Row(
                  children: [
                    for (final (key, label) in [
                      ('small', 'Small'),
                      ('medium', 'Med'),
                      ('large', 'Large'),
                      ('xlarge', 'XL'),
                    ]) ...[
                      Expanded(
                        child: _ChoiceChip(
                          label: label,
                          isSelected: settings.clockSize == key,
                          accentColor: settings.clockColor,
                          onTap: () {
                            onUpdateSettings(settings.copyWith(clockSize: key));
                          },
                        ),
                      ),
                      if (key != 'xlarge') const SizedBox(width: 6),
                    ],
                  ],
                ),
              ],
            ),
          ),
          const SizedBox(height: 12),

          // 4. Clock Position
          SettingCard(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                const Text(
                  'Clock Position',
                  style: TextStyle(color: Colors.white, fontSize: 16, fontWeight: FontWeight.bold),
                ),
                const SizedBox(height: 12),
                Row(
                  children: [
                    for (final (key, label) in [
                      ('top', 'Top'),
                      ('center', 'Center'),
                      ('bottom', 'Bottom'),
                    ]) ...[
                      Expanded(
                        child: _ChoiceChip(
                          label: label,
                          isSelected: settings.clockPosition == key,
                          accentColor: settings.clockColor,
                          onTap: () {
                            onUpdateSettings(settings.copyWith(clockPosition: key));
                          },
                        ),
                      ),
                      if (key != 'bottom') const SizedBox(width: 6),
                    ],
                  ],
                ),
              ],
            ),
          ),
          const SizedBox(height: 20),

          // Reset to Defaults
          OutlinedButton.icon(
            style: OutlinedButton.styleFrom(
              foregroundColor: Colors.redAccent,
              side: const BorderSide(color: Colors.redAccent),
              shape: RoundedRectangleBorder(
                borderRadius: BorderRadius.circular(14),
              ),
              padding: const EdgeInsets.symmetric(vertical: 14),
            ),
            icon: const Icon(Icons.restart_alt),
            label: const Text('Reset All Settings to Defaults'),
            onPressed: () {
              onUpdateSettings(const ClockSettings());
            },
          ),
          const SizedBox(height: 24),
        ],
      ),
    );
  }
}

class _ChoiceChip extends StatelessWidget {
  final String label;
  final bool isSelected;
  final Color accentColor;
  final VoidCallback onTap;

  const _ChoiceChip({
    required this.label,
    required this.isSelected,
    required this.accentColor,
    required this.onTap,
  });

  @override
  Widget build(BuildContext context) {
    return GestureDetector(
      onTap: onTap,
      child: Container(
        height: 42,
        decoration: BoxDecoration(
          color: isSelected ? accentColor : Colors.white10,
          borderRadius: BorderRadius.circular(12),
        ),
        alignment: Alignment.center,
        child: Text(
          label,
          style: TextStyle(
            color: isSelected ? Colors.black : Colors.white,
            fontWeight: isSelected ? FontWeight.bold : FontWeight.w500,
            fontSize: 12,
          ),
        ),
      ),
    );
  }
}
